main.go

```diff
@@ -36,26 +35,68 @@ func echo(w http.ResponseWriter, r *http.Request) {
 		mt, message, err := c.ReadMessage()
 		if err != nil {
 			log.Println("read:", err)
-			if strings.Contains(string(err.Error()), "websocket") {
-				fmt.Println("Team 1 score: " + strconv.Itoa(goalsOne) + " Team 2 score: " + strconv.Itoa(goalsTwo))
-			}
 			break
 		}
-		log.Printf("recv: %s", message)
-		if strings.Contains(string(message), "INITIAL") {
-			_ = c.WriteMessage(mt, json.RawMessage("{\"start\": \"1\" }"))
+
+		var dispatcherMsg messages.DispatcherReqMsg
+		er := json.Unmarshal([]byte(message), &dispatcherMsg)
+
+		if er != nil {
+			log.Println(err)
 		}
 
-		if strings.Contains(string(message), "goal") {
-			if strings.Contains(string(message), "1") {
-				goalsOne++
+		if dispatcherMsg.MsgType == "INITIAL" {
+			message := messages.DispatcherResMsg{
+				GameId: dispatcherMsg.TableId,
 			}
+			msg, err := json.Marshal(message)
 
-			if strings.Contains(string(message), "2") {
-				goalsTwo++
+			if err != nil {
+				log.Fatalln(err)
 			}
+
+			c.WriteMessage(mt, msg)
```

Comment:
You log fact that error was returned, but this code will continue and send nil via websocket. 

I think it's just mistake, so I suggest to add return statement like at lines 79-82:
```
if err != nil {
    log.Print("upgrade:", err)
    return
}
```